spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaLog.scala

```diff
@@ -161,13 +167,19 @@ private[sharing] object RemoteDeltaLog {
 class RemoteSnapshot(
     tablePath: Path,
     client: DeltaSharingClient,
-    table: DeltaSharingTable) extends Logging {
+    table: DeltaSharingTable,
+    versionOf: Option[Long] = None) extends Logging {
 
   protected def spark = SparkSession.active
 
   lazy val (metadata, protocol, version) = {
-    val tableMetadata = client.getMetadata(table)
-    (tableMetadata.metadata, tableMetadata.protocol, tableMetadata.version)
+    if (versionOf.isEmpty) {
+      val tableMetadata = client.getMetadata(table)
+      (tableMetadata.metadata, tableMetadata.protocol, tableMetadata.version)
+    } else {
+      val tableFiles = client.getFiles(table, Nil, Some(1L), versionOf)
+      (tableFiles.metadata, tableFiles.protocol, tableFiles.version)
```

Comment:
Question: Consider adding a comment that we request only 1 file for efficiency?

Reply:
added comment.